src/Illuminate/Support/Collection.php

```diff
@@ -54,6 +54,21 @@ public function all()
     }
 
     /**
+     * Get the average value of a given key.
+     *
+     * @param  string|null  $key
+     * @return mixed
+     */
+    public function avg($key = null)
+    {
+        if ($this->count() == 0) {
```

Comment:
Unless I assumed wrong, are you not checking if the $count variable is the same as what is contained in $this->count()?